Add Interactive Fun Section Page with Fact Cards and Mini Quiz

SMITAGADE opened the pull request:
closes #96 
Description:
This PR introduces a Fun Section page to the Skywatch Alerts app, aimed at engaging users with interactive content related to science and weather.

Features:

Rotating Fun Fact Cards: Displays 3 random facts at a time, updating every 8 seconds.

Mini Quiz: Users can answer a fun science/astronomy question. Correct answers trigger a confetti celebration.

Responsive Design: Fully functional on mobile and desktop.

Animations: Fade-in animation for cards and quiz, hover effects for interactivity.

Navigation: Back to Home button for easy navigation.

Files Added/Modified:

src/components/FunSection.tsx – Main interactive component.

src/app/fun-section/page.tsx – Fun Section page.

Optional: globals.css updated for fade-in animation.

How to Test:

Start the development server: npm run dev.

Open the Home page and click the Fun Section button.

Verify:

Fact cards rotate every 8 seconds.

Quiz shows options and triggers confetti on the correct answer.

Back button navigates to Home.

Page looks good on mobile and desktop.

Notes:

Uses react-confetti package; make sure it’s installed: npm install react-confetti.

Designed to be easily extensible with additional quizzes, polls, or animations.
<img width="1581" height="568" alt="Screenshot 2025-10-20 121627" src="https://github.com/user-attachments/assets/90150842-9387-40bf-a719-19699efe49d6" />
<img width="1846" height="812" alt="Screenshot 2025-10-20 121649" src="https://github.com/user-attachments/assets/824eba83-89d8-4d41-8ed8-91a41532a43f" />

SMITAGADE committed:
Add interactive Fun Section page with rotating facts and mini quiz

SMITAGADE committed:
Add interactive Fun Section page with rotating facts and mini quiz

SMITAGADE commented:
Please review and merge this pr ☺️

Dnyaneshpise commented:
@SMITAGADE The current UI doesn’t fully match the app’s theme, and the content could be made more dynamic to align with flight-related content.

SMITAGADE committed:
Enhance Fun Section: Flight-themed UI, quiz styling, and animations

SMITAGADE commented:
@Dnyaneshpise Here's a summary of all the changes I made

Refactored into Components:

The single FunSection component was broken down into smaller, manageable components:

FunSection.tsx (Main layout)

FlightQuiz.tsx (All quiz logic)

QuickChallengeCard.tsx (For interactive mini-games)

DailyStreak.tsx (New self-contained feature)

Upgraded: "Fun Facts" to "Quick Challenges"

Replaced the static, rotating "Fun Facts" with interactive QuickChallengeCard components.

Added three types of challenges:

trueFalse: A simple True/False question.

scramble: An "Unscramble the Airport Code" mini-game.

imageQuiz: An "Identify This!" challenge using images (e.g., airline tails, aircraft silhouettes).

Upgraded: "Flight Quiz" (Major Overhaul)

Converted the single-question quiz into a full multi-question quiz engine.

Added a 15-second per-question timer to create urgency.

Implemented a local storage High Score tracker to encourage replayability.

Added a "Restart Quiz" button and a final score summary screen.

New Feature: "Daily Streak"

Added a new DailyStreak component that appears in the corner.

It uses local storage to track the user's consecutive daily visits, adding a powerful gamification element to encourage return engagement.

New Feature: Image Support

Integrated images from the /public folder to support the new imageQuiz challenge type.

SMITAGADE commented:
Can you merge this pr😊 @Dnyaneshpise
